docs/architecture/adr-010-unified-cli-arguments-hermes.md

```diff
@@ -0,0 +1,184 @@
+
+* Only long flags are used in order to avoid having nonintuitive flags or conflicting flags.
+* Any parameter ending with `_id` should have the `_id` removed from the flag to shorten it. For example the flag for `chain_id` should only be `chain`.
+* Flags which can be shorten and still be meaningful should be shorten. This is done for `connection`, `channel` and `sequence`, which become respectively `conn`, `chan` and `seq`.
+* For the channel and connection creation CLIs, the objects at the two ends are prefixed by `--a-` and `--b-`.
+* All the specifiers for `chain`, `connection`, `channel` and `port` flags are prefixed. These specifiers are `host`, `reference`, `a`, `b` and `counterparty`.
+
+The following commands are implemented, with the binary name `hermes` omitted:
+
+### Hermes global flags
+
+* `hermes --config <CONFIG> <COMMAND>`
+
+* `hermes --json <COMMAND>`
+
+### Commands for clients
+
+* `create client --host-chain <HOST_CHAIN_ID> --reference-chain <REFERENCE_CHAIN_ID>`
+    * Optional: `[--clock-drift <CLOCK_DRIFT>] [--trust-threshold <TRUST_THRESHOLD>] [--trusting-period <TRUSTING_PERDIOD>]`
+
+* `update client --host-chain <HOST_CHAIN_ID> --host-client <HOST_CLIENT_ID>`
+    * Optional: `[--height <REFERENCE_HEIGHT>] [--trusted-height <REFERENCE_TRUSTED_HEIGHT>]`
+
+* `upgrade client --host-chain <HOST_CHAIN_ID> --host-client <HOST_CLIENT_ID>`
+
+* `upgrade clients --reference-chain <REFERENCE_CHAIN_ID>`
+    * Optional: `[--host-chain <HOST_CHAIN_ID>]`
+
+### Create a connection
+
+* `create connection --a-chain <A_CHAIN_ID> --b-chain <B_CHAIN_ID>`
+    * Optional: `[--delay <DELAY>]`
+
+* `create connection --a-chain <A_CHAIN_ID> --a-client <A_CLIENT_ID> --b-client <B_CLIENT_ID>`
+    * Optional: `[--delay <DELAY>]`
+
+### Create a channel
+
+* `create channel --a-chain <A_CHAIN_ID> --b-chain <B_CHAIN_ID> --a-port <A_PORT_ID> --b-port <B_PORT_ID>`
+    * Optional: `[--chan-version <VERSION>] [--new-client-conn] [--order <ORDER>]`
+
+* `create channel --a-chain <A_CHAIN_ID> --a-conn <A_CONNECTION_ID> --a-port <A_PORT_ID> --b-port <B_PORT_ID>`
+    * Optional: `[--chan-version <VERSION>] [--order <ORDER>]`
+
+### Commands for keys
+
+* `keys add --chain <CHAIN_ID> --key-file <KEY_FILE> --mnemonic-file <MNEMONIC_FILE>`
+    * Optional: `[--hd-path <HD_PATH>] [--key-name <KEY_NAME>]`
+
+* `keys balance --chain <CHAIN_ID>`
+    * Optional: `[--key-name <KEY_NAME>]`
+
+* `keys delete --chain <CHAIN_ID> --all`
+
+* `keys delete --chain <CHAIN_ID> --key-name <KEY_NAME>`
+
+* `keys list --chain <CHAIN_ID>`
+
+### Listen
+
+* `listen --chain <CHAIN_ID>`
+    * Optional: `[--event <EVENT>]`
+
+### Misbehaviour
+
+* `misbehaviour --chain <CHAIN_ID> --client <CLIENT_ID>`
+
+### Start the relayer in multi-chain mode
+
+* `start`
+    * Optional: `[--full-scan]`
+
+### Clear objects
+
+* `clear packets --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+
+### Queries
```

Comment:
can we order the queries in the same way as above for create/update, i.e. clients, connections, channels, packets.